tests/docker/ducktape-deps/polaris

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -e
+update-java-alternatives -s java-1.21.0-openjdk-amd64
```

Comment:
curious if we need to use `update-java-alternatives` instead of setting `JAVA_HOME` since update alternatives switches it at a system level

Reply:
i didn't want the script to mutate the system settings every time it is run

Reply:
In the current form, it does mutate, right? I thought JAVA_HOME is the more idiomatic way of doing this with multiple JDKs (not suggesting to do it in the PR, ok to go in, just out of my own curiosity).

Reply:
this is limited to the polaris layer in a Docker image

Reply:
right, what I meant by mutation is since `update-java-alternatives` actually makes a change in the FS by updating symlinks, its probably better to do something like..

JAVA_HOME=/path/to/21 ./gradlew --no-daemon --info shadowJar  

instead of updating symlinks twice, nbd was just brainstorming.